### PR TITLE
[HttpKernel] Fix error logger when stderr is redirected to /dev/null

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | n/a 
| License       | MIT
| Doc PR        | n/a

The HttpKernel Logger is meant to be used as a last resort logging mechanism when no logger has been explicitly configured (Monolog is not a dependency for instance).

For small apps, that can be more than enough.

But under some circumstances, it does not work. When you are using PHP-FPM, `stderr` is ignored by default (`catch_workers_output` is `false`) and so, logs are ignored as well. There is no issue with the official PHP Docker image as the setting has been explicitly set to `true`. Not an issue with Symfony CLI as well, as we also change the setting. Not a problem either with the PHP built-in server as it does not use PHP FPM anyway.

But, in many other places, where the setting has its default value, logs are lost (as you can imagine, it happened to me). As this feature is meant to be a fallback, I think it should always work, or at least, we need to make everything possible to make it work out of the box; that's why I've considered it a bug and hence a PR on 3.4.

This PR changes the default value for the output to `null`, which uses `error_log()` instead of `stderr` to log errors. Why is it better? The output of `error_log()` is controllable by the `error_logs` PHP ini setting and it is well understood by everyone (the default configuration should always work well); so it should work in most/more cases.

The other change (to be discussed) is to also log messages at the `ERROR` level and not just the `CRITICAL` ones.

/cc @dunglas

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Log/Logger.php [hunk 1]
```diff
@@ -37,10 +37,10 @@ class Logger extends AbstractLogger
     private $formatter;
     private $handle;
 
-    public function __construct($minLevel = null, $output = 'php://stderr', callable $formatter = null)
+    public function __construct($minLevel = null, $output = null, callable $formatter = null)
     {
         if (null === $minLevel) {
-            $minLevel = 'php://stdout' === $output || 'php://stderr' === $output ? LogLevel::CRITICAL : LogLevel::WARNING;
+            $minLevel = 'php://stdout' === $output || 'php://stderr' === $output ? LogLevel::ERROR : LogLevel::WARNING;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Log/Logger.php:43]: `null === $output || ...`

File: src/Symfony/Component/HttpKernel/Log/Logger.php [hunk 2]
```diff
@@ -58,8 +58,10 @@ public function __construct($minLevel = null, $output = 'php://stderr', callable
 
         $this->minLevelIndex = self::$levels[$minLevel];
         $this->formatter = $formatter ?: [$this, 'format'];
-        if (false === $this->handle = \is_resource($output) ? $output : @fopen($output, 'a')) {
-            throw new InvalidArgumentException(sprintf('Unable to open "%s".', $output));
+        if ($output) {
+            if (false === $this->handle = \is_resource($output) ? $output : @fopen($output, 'a')) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Log/Logger.php:62]: we can combine both "if"s

**Review by nicolas-grekas — APPROVED** [review]
LGTM, here are some minor things.

**Review by dunglas — APPROVED** [review]
lgtm

**Comment by Naktibalda** [comment]
I really don't like this change because php-fpm prepends `NOTICE: PHP message: ` to all lines when error_log is used.

I made logger work the way it did before by adding this section to config/services.yaml
```yaml
    logger:
        class: Symfony\Component\HttpKernel\Log\Logger
        arguments:
            $output: 'php://stderr'
```

**Comment by stof** [comment]
@Naktibalda if your PHP-FPM is configured to preserve stderr, that's a valid configuration (if you still don't want to use a more powerful logger like monolog).
But as explained in the description, the goal of this change is that the fallback logger (when not configuring any `logger` service) does work whatever the PHP-FPM config.

Having a `NOTICE: PHP message: ` prefix is better than having no logs.

**Pull request merged**